modules/apps/portlet-configuration/portlet-configuration-web/src/main/java/com/liferay/portlet/configuration/web/internal/display/context/PortletConfigurationPermissionsDisplayContext.java

```diff
@@ -367,6 +367,8 @@ public SearchContainer<Role> getRoleSearchContainer() throws Exception {
 		RoleSearchTerms searchTerms =
 			(RoleSearchTerms)roleSearchContainer.getSearchTerms();
 
+		setKeywords(searchTerms.getKeywords());
```

Comment:
or you can use the set to fill with this value OR, if null, the one from the request so you don't need the get method but you can use directly _keywords

I don't know if it works but you should try because I don't like that here we are settings something but from the get we can receive a different value